specs/online_order_spec.rb

```diff
@@ -3,69 +3,145 @@
 require 'minitest/skip_dsl'
 
 # TODO: uncomment the next line once you start wave 3
-# require_relative '../lib/online_order'
+require_relative '../lib/online_order'
+require_relative '../lib/customer'
+
 # You may also need to require other classes here
 
 # Because an OnlineOrder is a kind of Order, and we've
 # already tested a bunch of functionality on Order,
 # we effectively get all that testing for free! Here we'll
 # only test things that are different.
 
-xdescribe "OnlineOrder" do
+describe "OnlineOrder" do
   describe "#initialize" do
     it "Is a kind of Order" do
       # Check that an OnlineOrder is in fact a kind of Order
-
       # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      # online_order = Grocery::OnlineOrder.new(Grocery::Customer.new("123","sgash@email.com","122323"))
+      online_order = Grocery::OnlineOrder.new(1234, { "banana" => 1.99, "cracker" => 3.00 },
+      Grocery::Customer.new("123","sgash@email.com","122323"), :pending)
+
+      online_order.must_be_kind_of Grocery::Order
+      online_order.must_be_kind_of Grocery::OnlineOrder
     end
 
     it "Can access Customer object" do
-      # TODO: Your test code here!
+      # online_order = Grocery::OnlineOrder.new(Grocery::Customer.new("123","sgash@email.com","122323"))
+      online_order = Grocery::OnlineOrder.new(1234, { "banana" => 1.99, "cracker" => 3.00 },
+      Grocery::Customer.new("123","sgash@email.com","122323"), :pending)
+
+      online_order.customer.must_be_kind_of Grocery::Customer
     end
 
     it "Can access the online order status" do
-      # TODO: Your test code here!
+      # online_order = Grocery::OnlineOrder.new(Grocery::Customer.new("123","sgash@email.com","122323"),:pending )
+      online_order = Grocery::OnlineOrder.new(1234, { "banana" => 1.99, "cracker" => 3.00 },
+      Grocery::Customer.new("123","sgash@email.com","122323"), :pending)
+      online_order.status.must_equal :pending
     end
   end
 
   describe "#total" do
-    it "Adds a shipping fee" do
-      # TODO: Your test code here!
+    it "Adds a shipping fee" do # def initialize (id, products, customer, status = :pending
+      online_order = Grocery::OnlineOrder.new(1234, { "banana" => 1.99, "cracker" => 3.00 },
+      Grocery::Customer.new("123","sgash@email.com","122323"), :pending)
+
+      prev_total =  Grocery::Order.new(231,{ "banana" => 1.99, "cracker" => 3.00 })
+      online_order.total.must_equal prev_total.total + 10
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      online_order = Grocery::OnlineOrder.new(1234, { "banana" => 1.99, "cracker" => 3.00 },
+      Grocery::Customer.new("123","sgash@email.com","122323"), :pending)
```

Comment:
I don't know that this test code matches the name of the test. Your description talks about "if there are no products", but then you create an `OnlineOrder` with products. What if instead you did something like this:

```ruby
online_order = Grocery::OnlineOrder.new(1234, {}, customer, :pending)
online_order.total.must_equal 0
```